### ISSUE TITLE
Azure blobs - setting AzureBlobsStorageDriver  `host` does not work for non-emulated requests

### ISSUE THREAD

**Issue by mkcello96** [apache/libcloud#1551]
## Summary

I'm using libcloud to connect to Azure Gov, so I need to change `host` in my AzureBlobsStorageDriver connection object (which uses `<account>.blob.core.usgovcloudapi.net` instead of the default `<account>.blob.core.windows.net`). This does not work - the call fails with `Unexpected status code: 400`.

This broke after https://issues.apache.org/jira/browse/LIBCLOUD-1033 was merged ~2 years ago.

## Detailed Information

https://issues.apache.org/jira/browse/LIBCLOUD-1033 adds support for Azurite, which largely assumes libcloud users would be setting `host` only if using an Azure emulator. Thus, it will always set an `account_prefix`=`key` IFF a `host` is specified, which will break non-emulated Azure calls ([see how a list containers call has a separate Request URI for Emulated calls](https://docs.microsoft.com/en-us/rest/api/storageservices/list-containers2)). 
A workaround for those not using emulators was proposed [here](https://github.com/apache/libcloud/blame/fe4f81b5a31a8f470ff6ba5af7451cccfc423c2b/libcloud/storage/drivers/azure_blobs.py#L167), to set the key to '' (empty string). **This workaround also does not work** - the call also fails with `Unexpected status code: 400`.

After a lengthy debugging session I figured out why the proposed workaround does not work: Setting `key` to the empty string does not prevent `account_prefix` from being set (it also gets set to the empty string). Further, doing this adds an extra `/` to the Request URI ([code here](https://github.com/apache/libcloud/blame/fe4f81b5a31a8f470ff6ba5af7451cccfc423c2b/libcloud/storage/drivers/azure_blobs.py#L193)), which will cause the Azure request to fail. Setting `key=None` fixes this issue.
However that is not the only issue with the proposed workaround - not having a `key` also prevents Azure authentication from working. Therefore I had to add a `driver.connection.user_id = key_variable` statement after the AzureBlobsStorageDriver object was initialized to fix this.

TL;DR - my current workaround is to do this:
```
AzureBlobsStorageDriver(key=None, secret=<secret>, host=<storageAccount>.blob.core.usgovcloudapi.net )
driver.connection.user_id = <storageAccount>
```

My fix recommendation is that an `emulated` boolean flag parameter be added to the AzureBlobsStorageDriver initializer, which will only set the `account_prefix` if it's True. This should also allow the `key` parameter to be set with the `host`.

OS: Mac OS
Python version: 3.6.8
Libcloud version: 3.3.1

**Comment by Kami** [comment]
Thanks for reporting this.

@c-w Any chance you will be able to look at this in the near future?

**Comment by c-w** [comment]
@Kami Yes, happy to look into it.

**Comment by c-w** [comment]
@mkcello96 I have a patch for this but I don't currently have access to an Azure Government subscription to verify the fix end-to-end. Could you instantiate the driver at version 505894fa as shown below and let me know if your issue is resolved? Thanks in advance!

```py
from libcloud.storage.types import Provider
from libcloud.storage.providers import get_driver

cls = get_driver(Provider.AZURE_BLOBS)

driver = cls(key='your storage account name',
             secret='your access key',
             host='blob.core.usgovcloudapi.net')
```

**Comment by mkcello96** [comment]
Hey @c-w thanks so much for putting a fix in for this. I'm getting this error now:

`
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/mkersh/Documents/libcloud/libcloud/storage/drivers/azure_blobs.py", line 221, in __init__
    port=port, **kwargs)
  File "/Users/mkersh/Documents/libcloud/libcloud/common/base.py", line 1008, in __init__
    self.connection = self.connectionCls(*args, **conn_kwargs)
  File "/Users/mkersh/Documents/libcloud/libcloud/storage/drivers/azure_blobs.py", line 191, in __init__
    super(AzureBlobsConnection, self).__init__(*args, **kwargs)
TypeError: __init__() got multiple values for argument 'host'
`

I confirmed this was on the locally-built branch:
`
python3 -c "import libcloud;  print(libcloud.__version__)"
3.3.2-dev
`

**Comment by c-w** [comment]
Thanks @mkcello96. This should be resolved as of b16eb28.

**Comment by mkcello96** [comment]
Working for me now @c-w,  thanks!
